python/lsst/ip/diffim/getTemplate.py

```diff
@@ -702,6 +693,203 @@ def run(self, coaddExposures, bbox, wcs):
             raise RuntimeError("CoaddPsf could not be constructed")
 
         templateExposure.setPsf(coaddPsf)
-        templateExposure.setFilterLabel(coaddPatch.getFilterLabel())
-        templateExposure.setPhotoCalib(coaddPatch.getPhotoCalib())
+        templateExposure.setFilterLabel(coaddExposure.getFilterLabel())
+        templateExposure.setPhotoCalib(coaddExposure.getPhotoCalib())
         return pipeBase.Struct(outputExposure=templateExposure)
+
+
+class GetDcrTemplateConnections(GetTemplateConnections,
+                                dimensions=("instrument", "visit", "detector", "skymap"),
+                                defaultTemplates={"coaddName": "dcr",
+                                                  "warpTypeSuffix": "",
+                                                  "fakesType": ""}):
+    visitInfo = pipeBase.connectionTypes.Input(
+        doc="VisitInfo of calexp used to determine observing conditions.",
+        name="{fakesType}calexp.visitInfo",
+        storageClass="VisitInfo",
+        dimensions=("instrument", "visit", "detector"),
+    )
+    dcrCoadds = pipeBase.connectionTypes.Input(
+        doc="Input DCR template to match and subtract from the exposure",
+        name="{fakesType}dcrCoadd{warpTypeSuffix}",
+        storageClass="ExposureF",
+        dimensions=("tract", "patch", "skymap", "band", "subfilter"),
+        multiple=True,
+        deferLoad=True
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+        self.inputs.remove("coaddExposures")
+
+
+class GetDcrTemplateConfig(GetTemplateConfig,
+                           pipelineConnections=GetDcrTemplateConnections):
+    numSubfilters = pexConfig.Field(
+        doc="Number of subfilters in the DcrCoadd.",
+        dtype=int,
+        default=3,
+    )
+    effectiveWavelength = pexConfig.Field(
+        doc="Effective wavelength of the filter.",
+        optional=False,
+        dtype=float,
+    )
+    bandwidth = pexConfig.Field(
+        doc="Bandwidth of the physical filter.",
+        optional=False,
+        dtype=float,
+    )
+
+    def validate(self):
+        if self.effectiveWavelength is None or self.bandwidth is None:
+            raise ValueError("The effective wavelength and bandwidth of the physical filter "
+                             "must be set in the getTemplate config for DCR coadds. "
+                             "Required until transmission curves are used in DM-13668.")
+
+
+class GetDcrTemplateTask(GetTemplateTask):
+    ConfigClass = GetDcrTemplateConfig
+    _DefaultName = "getDcrTemplateTask"
```

Comment:
My fault for including the `Task` in the original getTemplate Tasks. Should be `_DefaultName = "getDcrTemplate"`